enhancements/installer/cluster-api-based-installations.md

```diff
@@ -0,0 +1,557 @@
+
+##### Bootstrap Resources
+
+The Bootstrap Machine is created like the other control plane nodes, by defining a CAPI Machine resource.
+Accordingly, the bootstrap machine can be deleted by simply deleting the object. This works for all platforms:
+
+```go
+	if err := c.Delete(ctx, &clusterv1.Machine{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      capiutils.GenerateBoostrapMachineName(metadata.InfraID),
+			Namespace: capiutils.Namespace,
+		},
+	});
+```
+
+Further work needs to be done to determine how to delete additional bootstrap resources, such as SSH security group rules
+and the bootstrap ignition S3 bucket. These could be deleted either through updating the relevant manifests or created &
+deleted out-of-band using hooks.
+
+### Risks and Mitigations
+
+While we do not expect these changes to introduce a significant security risk, we are working with product security teams
+to ensure they are aware of the changes and are able to review.
+
+
+### Drawbacks
+
+#### External/Upstream dependencies
+
+By depending on CAPI providers whose codebases live in a repository external to the Installer,
+the process for developing features and delivering fixes is more complex than in a monolothic repo.
+While the same could be true for the Installer Terraform dependency; the CAPI providers will
+be more actively developed than their Terraform counterparts. Furthermore, it will be necessary
+to ensure that the CAPI providers used by the Installer match the version of those in the payload.
+
+While this external dependency is a significant drawback, it is not unique to this design
+and is common throughout OpenShift (e.g. any time the API or library-go must be updated
+before being vendored into a component). To minimize the devex friction, we will focus
+on documenting a workflow for developing providers while working with the Installer.
+
+Additionally, we will explore designs to solve or mitigate these issues.
+
+#### Disk space and memory
+
+> NOTE: All tests are performed on a MacOS system at the time of writing.
+
+With the introduction of the Local Control Plane, new required components are being run during cluster creation.
+
+- `api-server` uses ~280MB and `etcd` ~52MB
+- Cluster API
+  - Core controller uses ~28MB
+  - Infrastructure controller (AWS, in testing) uses ~35MB
+
+When compared to Terraform, the runtime memory footprint is well-within the current usare or limitations (TODO: provide Terraform benchmark); the disk footprint increases by what's required of `etcd` which is around 128MB while creating an OpenShift cluster on AWS.
+
+#### Binary size
+
+> NOTE: All tests are performed on a MacOS system at the time of writing.
+
+The `openshift-install` resulting binary size, with Cluster API bundled binaries reaches ~515MB, of which, the bundled binaries account 228MB. By comparison, the binary available for download on `console.redhat.com` for MacOS, once upacked, is 732MB.
```

Comment:
How complete is our platform coverage in this measurement against where we were with TF? Just interested to know if we only have half the platforms and expect provider size to be mostly uniform are we going to be smaller or larger than current binaries.

Reply:
IIRC the measurement was taken with all providers - 2 (libvirt and alicloud), I'll update

Reply:
added a note to clarify